### PR TITLE
Support get account detail query

### PR DESCRIPTION
**Pull request by MizukiSonoko**
Re-Open 🙇‍♀️
https://github.com/hyperledger/iroha/pull/784

### CODE REVIEW

File: irohad/model/queries/get_account_detail.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_GET_ACCOUNT_DETAIL_HPP
+#define IROHA_GET_ACCOUNT_DETAIL_HPP
+
+#include <model/query.hpp>
```
Comment (l4l) [inline irohad/model/queries/get_account_detail.hpp]: Should be in the ".."

File: irohad/model/queries/responses/account_detail_response.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_ACCOUNT_DETAIL_RESPONSE_HPP
+#define IROHA_ACCOUNT_DETAIL_RESPONSE_HPP
+
+#include <vector>
```
Comment (l4l) [inline irohad/model/queries/responses/account_detail_response.hpp]: Maybe string instead?

File: shared_model/backend/protobuf/query_responses/proto_account_detail_response.hpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            const std::string &detail() const override {
+              return *detail_;
+            }
+
+        private:
+            template <typename T>
+            using Lazy = detail::LazyInitializer<T>;
+
+            const Lazy<const iroha::protocol::AccountDetailResponse &>
+              accountDetailResponse_;
```
Comment (l4l) [inline shared_model/backend/protobuf/query_responses/proto_account_detail_response.hpp]: Should be in snake_case

File: shared_model/backend/protobuf/query_responses/proto_account_detail_response.hpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                    return accountDetailResponse_->detail();
+                }) {}
+
+            AccountDetailResponse(const AccountDetailResponse &o)
+              : AccountDetailResponse(o.proto_) {}
+
+            AccountDetailResponse(AccountDetailResponse &&o)
+              : AccountDetailResponse(std::move(o.proto_)) {}
+
+            const std::string &detail() const override {
```
Comment (l4l) [inline shared_model/backend/protobuf/query_responses/proto_account_detail_response.hpp]: Shdouln't it be `interface::types::DetailType`?

File: shared_model/interfaces/query_responses/account_detail_response.hpp [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_ACCOUNT_DETAIL_RESPONSE_HPP
+#define IROHA_SHARED_MODEL_ACCOUNT_DETAIL_RESPONSE_HPP
+
+#include <new>
```
Comment (l4l) [inline shared_model/interfaces/query_responses/account_detail_response.hpp]: It's included by default afair. Is there a compilation error without it?

File: shared_model/interfaces/query_responses/account_detail_response.hpp [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+              return detail() == rhs.detail();
+            }
+
+            /**
+             * Makes old model.
+             * @return An allocated old model of account asset response.
+             */
+            OldModelType *makeOldModel() const override {
+              OldModelType *oldModel = new OldModelType();
+              const auto vs = detail();
```
Comment (l4l) [inline shared_model/interfaces/query_responses/account_detail_response.hpp]: Redundant var?

File: test/module/irohad/model/converters/pb_query_responses_test.cpp [hunk 7]
```diff
@@ -92,6 +92,30 @@ TEST(QueryResponseTest, AccountAsset) {
...
+ */
+TEST(QueryResponseTest, AccountDetailResponse) {
+  model::converters::PbQueryResponseFactory pb_factory;
+
+  std::string detail = "{}";
+
+  model::AccountDetailResponse account_detail_response;
+  account_detail_response.detail = detail;
+
+  auto shrd_aar = std::make_shared<decltype(account_detail_response)>(
```
Comment (l4l) [inline test/module/irohad/model/converters/pb_query_responses_test.cpp]: Name is a quite strange abbreviation, google style prohibit such ones

**Comment by neewy** [comment]
Please reopen PR, restore the state of develop before the merge and have an additional review from any maintainer 🙇

**Pull request merged**